spec/lib/classes/organisations_user_builder_spec.rb

```diff
@@ -0,0 +1,65 @@
+require "rails_helper"
+
+  # :organisations_user: {
+  #   :organisation_id,
+  #   :position,
+  #   user_attributes: {
+  #     :first_name,
+  #     :last_name,
+  #     :mobile,
+  #     :email
+  #   }
+
+describe OrganisationsUserBuilder do
+  let(:organisation) { create :organisation}
+  let(:user_attributes) do
+    FactoryBot.attributes_for(:user, :with_email, mobile:'+85252345678')
+  end
+  let(:organisations_user_params) do
+    FactoryBot.attributes_for(:organisations_user, organisation_id: "#{organisation.id}", position: "#{position}" , user_attributes: user_attributes)
+  end
+
+  let(:user) { create :user }
+  let(:current_user) { create :user }
+  let(:user1) { create :user, mobile: user_attributes[:mobile] }
+  let(:position) { 'Admin' }
+
+  let(:organisations_user_builder) { OrganisationsUserBuilder.new(organisations_user_params) }
+  let(:mobile) { '+85251111111' }
+  let!(:role) { create :charity_role }
+  let(:subject) { JSON.parse(response.body) }
+
+  context "initialization" do
+    it { expect(organisations_user_builder.instance_variable_get("@organisation_id")).to eql(organisation.id) }
+    it { expect(organisations_user_builder.instance_variable_get("@user_attributes")).to eql(user_attributes) }
+    it { expect(organisations_user_builder.instance_variable_get("@mobile")).to eql(user_attributes[:mobile]) }
+    it { expect(organisations_user_builder.instance_variable_get("@position")).to eql(position) }
+  end
+
+  context "build" do
+    it "adds new user if mobile does not exist and associates it with organisation" do
+      User.current_user = current_user
+      organisations_user_builder.build
+      expect(User.count).to eq(3)
+      expect(OrganisationsUser.count).to eq(1)
```

Comment:
Could you please try below code here?

expect{
 organisations_user_builder.build}
}.to change{User.count}.by(1).and change{OrganisationUser.count}.by(1)